### PR TITLE
Rename `current` amounts to `recent`, etc. Deprecate old.

### LINKED ISSUE #2100: Rename purse `getCurrentAmount` to `getRecentAmount` and deprecate `getCurrentAmount`
**What is the Problem Being Solved?**

The notion of a "current" amount is not very useful when purses typically are in separate vats, since the returned value may be stale after the turn that sent it has passed.  If we relax "current" instead to be "recent", it sets user expectations to be just useful information about past balances instead of some kind of instantaneous truth.  This is especially true because `getCurrentAmountNotifier` is already lossy.

Relaxing the name allows us to implement windows into remote purses with consistent balances whose observation may be delayed.

**Description of the Design**

1. Change `getCurrentAmount` to be `getRecentAmount` and `getCurrentAmountNotifier` to be `getRecentAmountNotifier`.
2. Deprecate the old names.

**Security Considerations**

Consumers of the purse API must not rely on the instantaneity of the reported balance.  We hope that by changing the names, this requirement will be better communicated.

**Test Plan**

TBD.

### PR DESCRIPTION
**Pull request by erights**
Fixes #2100 

#documentation-branch: current-patch-21-12-20

### CODE REVIEW

File: packages/ERTP/src/types.js
Comment (erights) [inline packages/ERTP/src/types.js:266]: Note: I need advice here as explained in the TODO comment.

File: packages/dapp-svelte-wallet/api/src/types.js
Comment (erights) [inline packages/dapp-svelte-wallet/api/src/types.js:98]: Note: I need advice here as explained in the TODO comment.

**Comment by erights** [comment]
https://github.com/Agoric/documentation/pull/408 should fix the remaining failure.

File: packages/ERTP/src/types.js
Comment (katelynsills) [reply packages/ERTP/src/types.js:266]: The JSDoc example has it applied to a single function. Would it work if you pulled this function out? https://jsdoc.app/tags-deprecated.html

Comment (erights) [reply packages/ERTP/src/types.js:266]: How would I pull this function out when it is part of an object that implements a declared interface? I just don't know jsdoc well enough to understand what kinds of refactoring I can do on these declarations.

File: packages/dapp-svelte-wallet/api/src/types.js
Comment (michaelfig) [reply packages/dapp-svelte-wallet/api/src/types.js:98]: I didn't see them used anywhere except in the Wallet UI, so I suspect the answer is "no", but I will leave it up to @katelynsills when she takes this on.

File: packages/ERTP/src/types.js
Comment (michaelfig) [reply packages/ERTP/src/types.js:266]: Just for the record: I can't figure out how to deprecate a single property.  Googling around implies it isn't possible with current JSDoc, as the `@deprecation` tag is just for a specific symbol, not a property.

File: packages/dapp-svelte-wallet/api/src/types.js
Comment (katelynsills) [reply packages/dapp-svelte-wallet/api/src/types.js:98]: IMO, we don't need to keep a deprecated currentAmountSlots

**Review by katelynsills — APPROVED** [review]
Looks good to me. Let's wait for companion documentation from @tyg before merging

**Comment by katelynsills** [comment]
Closing as we have decided not to make this change

**Pull request closed without merging**
